### ISSUE TITLE
Add oscars conda package

### ISSUE THREAD

**Issue by danielballan** [NSLS-II/lightsource2-recipes#205]
I've been manually installing `oscars==1.37.1.dev3` using pip for @dhidas.

We should make a conda package so it can be added to our 'analysis' conda metapackages and installed automatically everywhere.

**Comment by stuartcampbell** [comment]
Cool - @licode can you have a look at it ?

**Comment by licode** [comment]
Sure, I will add this.

**Comment by dhidas** [comment]
Hi,

Thanks.  I'm happy to do this with you guys.  If we can have a discussion at some point that might help me.  There are some complications with conda, namely compiler name mangling that differs from nvidia's nvcc which I haven't been able to reconcile so far.  If there are no GPU machines the solution is simple, which is just compile without it, but if there are it's worth it (a factor of > 1000 in many cases) to get this working with conda.  Of course, I can upload the manylinux1 binaries to pypi in any case...  but I'd like to know really the best course of action.

Also, I'm planning a major upgrade in the next ~1.5 weeks, maybe even earlier...

Cheers,

-Dean

**Comment by dhidas** [comment]
Hi,

Is there any chance we could quickly upgrade to the latest version: pip install oscars==2.0.2  ?

I don't know if it's a lot of effort or not.  Ideally it would be good to have it up in a week if that's not asking too much?

And if there is a way for me to make it easier for everyone in the future I'm all about that...

Cheers,

-Dean

**Comment by stuartcampbell** [comment]
@licode Have you made any progress?

**Comment by licode** [comment]
working on this. I find there are lots of c code in the github source code, and also calculations related to gpu and multithreading. This may be a bit challenging, and I will keep everyone updated.

**Comment by danielballan** [comment]
@dhidas Could you summarize for @licode what you told me a couple weeks ago about the dependency on a newer version of certain libraries than one finds on typical conda build systems (~CentOS 5)?

**Comment by dhidas** [comment]
The GPU tools need to be compiled with the nvcc compiler which is only available for cent os 6 or greater (and similar for other distros), so it won't exactly fit the manylinux1 standard, but perhaps would work well enough...or should I not go there?  I haven't found nvcc in any conda package yet, but they do distribute the libs in cudatoolkit.

To compile with the gpu I would normally:
make
python setup.py install

...however there are some issue when using conda, namely that nvcc name mangling was different from the conda gcc in some cases.  One could use nvcc to compile and link everything, but I didn't get that to play nice with setuptools, though I imagine it is possible.

There are nvidia docker images, but I haven't tried that at all yet.

**Comment by stuartcampbell** [comment]
I think this type of problem is something that we will need in more and more cases in the future, so it might be worth understanding the issues now and coming up with a 'recipe' for how we deal with using `nvcc` and cuda stuff... 

...now all we need is a volunteer to look into it 😈

**Comment by mrakitin** [comment]
The package itself was added. Closing this issue as resolved. If some specific issue is needed, a separate issue/PR should be opened, preferably to https://github.com/nsls-ii-forge/oscars-feedstock.
